public/validate-event-form.js

```diff
@@ -21,10 +21,20 @@
       description_en: elements.description_en.value,
       description_ar: elements.description_ar.value,
       eventPlace: elements.eventPlace.value,
-      imageUrl: elements.imageUrl.value,
       cost: elements.cost.value,
     };
 
+    // checks if a file is being uploaded
+    if (
+      document.getElementById('spinner').classList.value.indexOf('dn') === -1
+    ) {
+      return notValid('*Image did not finish upload');
```

Comment:
Okay, you need to disable the input button when the request is pending. The user should not be able to click

Reply:
yeah so when he clicks and chooses an image the spinner is gonna appear, and as long as the spinner is on the page if he clicks submit the validation won't let him make the api request

Reply:
Yes, this is not the correct behavior: I made an issue: https://github.com/foundersandcoders/OTP-Data-Entry/issues/100